src/internal/log/loggers.go

```diff
@@ -268,7 +268,9 @@ func InitBatchLogger(logFile string) func(err error) {
 		// closed log file.  (Needed 800us in testing.)
 		time.Sleep(5 * time.Millisecond)
 		if logFile != "" {
-			zap.L().Info(fmt.Sprintf("logfile retained at %v", logFile))
+			zap.L().Error(fmt.Sprintf("logfile retained at %q, due to error on exit: %v", logFile, err))
```

Comment:
I think this is going to double-print every error.  It's weird for the caller to not log their thing; it means they're treating the last error of the run different from other errors, and I don't think that's a particularly good idea.

All this code is supposed to do is make a decision as to whether or not to delete the log file; what actually goes into the log file is the caller's problem.  This is not the application running framework; that code should handle what to do about the final error.

Reply:
I would also accept something like wrapping the error you want to log here:

```go
type LogThisAtTheEnd struct { err error }
func (e *LogThisAtTheEnd) Error() string { return e.err.Error() } // implement builtin.error
...
log := new(logThisAtTheEnd)
if errors.As(err, &log) { 
   log.Error(...)
}
...
close := InitBatchLogger()
...
close(LogThisAtTheEnd(err))
``